### ISSUE TITLE
more composable/interoperable protos

### ISSUE THREAD

**Issue by pyramation** [Stride-Labs/stride#275]
Hello! I'm working on tooling to generate clients, and here to help with packaging protos in a canonical way that other chains are doing

For example,

- [./proto/stargaze](https://github.com/public-awesome/stargaze/tree/main/proto/stargaze)
- [./proto/osmosis](https://github.com/osmosis-labs/osmosis/tree/main/proto/osmosis)
- [./proto/akash](https://github.com/ovrclk/akash/tree/master/proto/akash)
- [./proto/cosmos](https://github.com/cosmos/cosmos-sdk/tree/main/proto)
- [./proto/juno](https://github.com/CosmosContracts/juno/tree/main/proto)

To make this more interoperable, can we move 

```
proto/epochs            -> proto/stride/epochs
proto/icacallbacks      -> proto/stride/icacallbacks
proto/interchainquery   -> proto/stride/interchainquery
proto/mint              -> proto/stride/mint
proto/records           -> proto/stride/records
proto/stakeibc          -> proto/stride/stakeibc
```

It looks like a pretty small change, but would be HUGE in terms of interoperability.

You may also want to consider the package names also, I think there are two packages: `stride` and `stridelabs`, not sure if it's good to consolidate or not depending on the use case.

cc @nghuyenthevinh2000

**Comment by pyramation** [comment]
Regarding package names, this type of thing will actually cause naming collisions (one is lowercase and one is pascal case, but both are global):

```proto
package stridelabs.stride.icacallbacks;
package Stridelabs.stride.epochs;
```

this is the source of https://github.com/cosmology-tech/stridejs/issues/1

**Comment by pyramation** [comment]
so let's look at a few packages and their names:

```proto
package Stridelabs.stride.stakeibc;
package stride.mint.v1beta1;
package stride.interchainquery;
package stridelabs.stride.icacallbacks;
package Stridelabs.stride.epochs;
```

May I suggest we do something like this:

```proto
package stride.stakeibc;
package stride.mint.v1beta1;
package stride.interchainquery;
package stride.icacallbacks;
package stride.epochs;
```

**Comment by nghuyenthevinh2000** [comment]
I have code for it but it is in my local. I am debugging script for buf generation.

**Comment by pyramation** [comment]
> I have code for it but it is in my local. I am debugging script for buf generation.

please see my PR https://github.com/Stride-Labs/stride/pull/282

do you think you can include the same changes I did for LCD endpoints? Those were also affected by using starport...

**Comment by pyramation** [comment]
thanks @nghuyenthevinh2000 — we can close my PR and use yours btw, I'm not a Go dev 😅

**Comment by asalzmann** [comment]
Thank you for these tips and both of your work on this! I think the ignite cli created these names in a funky way. Reviewing #283 now.

**Comment by sampocs** [comment]
Closed by #328 and #358 !
